libtenzir/builtins/formats/json.cpp

```diff
@@ -586,15 +618,10 @@ class default_parser final : public parser_base {
               view.data() + view.size(),
             };
             auto row = builder.record();
-            auto success
+            auto result
               = doc_parser{source, *dh}.parse_object(elem.value_unsafe(), row);
-            if (not success) {
-              // We already reported the issue.
+            if (result == doc_parser::result::failure_with_write) {
               builder.remove_last();
-              // It should be fine to continue here, because at least the array
-              // structure we are iterating is valid. That is ensured by the
-              // elem.error() check above
-              continue;
```

Comment:
Removing this `continue` doesn't do anything. What needs to happen if we fail here?

Reply:
I removed the continue precisely because it had no effect. I do see that keeping it may be clearer though, as in every other case we have to `break`.

Reply:
Is it fine to continue iterating the array, or is that another instance of the same problem?